homework/index.js

```diff
@@ -30,13 +30,175 @@
+    const table = createAndAppend('table', leftSideContent, {});
+    const tr1 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr1, { text: 'Repository: ' });
+    const td2 = createAndAppend('td', tr1, {});
+    createAndAppend('a', td2, {
+      id: 'repo-name',
+      text: `${firstSelOpt.text}`,
+      href: `${repos[firstSelElemIndex].html_url}`,
+      target: '_blank',
+    });
+    const tr2 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr2, { text: 'Description: ' });
+    createAndAppend('td', tr2, {
+      id: 'repo-description',
+      text: `${repos[firstSelElemIndex].description}`,
+    });
+    const tr3 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr3, { text: 'Fork: ' });
+    createAndAppend('td', tr3, { id: 'repo-fork', text: `${repos[firstSelElemIndex].forks}` });
+    const tr4 = createAndAppend('tr', table, {});
+    createAndAppend('td', tr4, { text: 'Updated: ' });
+    createAndAppend('td', tr4, {
+      id: 'repo-updated',
+      text: `${repos[firstSelElemIndex].updated_at}`,
+    });
+
+    function createRightSideInfos(contJsonData) {
+      const conts = contJsonData;
+      createAndAppend('h3', rightDiv, { text: 'Contributors' });
+      conts.forEach((_cont, index) => {
+        const contsInfo = createAndAppend('div', rightDiv, { class: 'cont-info' });
+        const contsInfoLeft = createAndAppend('div', contsInfo, { class: 'cont-info-left' });
+        const contsInfoRight = createAndAppend('div', contsInfo, { class: 'cont-info-right' });
+        createAndAppend('img', contsInfoLeft, {
+          src: `${conts[index].avatar_url}`,
+          alt: `${conts[index].login} avatar`,
+        });
+        createAndAppend('a', contsInfoLeft, {
+          text: `${conts[index].login}`,
+          href: `${conts[index].html_url}`,
+          target: '_blank',
+        });
+        createAndAppend('p', contsInfoRight, { text: `${conts[index].contributions}` });
+      });
+    }
+
+    const CONT_URL = repos[firstSelElemIndex].contributors_url;
+    fetchJSON(CONT_URL, (err, data) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        createRightSideInfos(data);
+      }
+    });
+
+    allOptions.addEventListener('change', () => {
+      const selOpt = getSelectedOption(allOptions);
+      const selElemIndex = getSelectedIndex(repos, selOpt);
+      document.getElementById('repo-name').innerHTML = `${selOpt.text}`;
+      document.getElementById('repo-name').setAttribute('href', `${repos[selElemIndex].html_url}`);
+      document.getElementById('repo-description').innerHTML = `${repos[selElemIndex].description}`;
+      document.getElementById('repo-fork').innerHTML = `${repos[selElemIndex].forks}`;
+      document.getElementById('repo-updated').innerHTML = `${repos[selElemIndex].updated_at}`;
+
+      function deleteRightSideInfos() {
```

Comment:
Now you're adding another nested function with its own nested functions. It's sort of hard to understand whether this is part of createHtml, addEventListener callback, or deleteRightSideInfos. Try to extract all these things in functions in the root scope.